src/python/turicreate/toolkits/image_analysis/image_analysis.py

```diff
@@ -192,6 +192,171 @@ def resize(image, width, height, channels=None, decode=False, resample="nearest"
             "Cannot call 'resize' on objects that are not either an Image or SArray of Images"
         )
 
+def resize_with_original_aspect_ratio(image, min_side_length=None, max_side_length=None, channels=None, decode=False, resample="nearest"):
+    """
+    Resizes the image or SArray of Images to a specific width or height while maintaining the aspect ratio
+
+    Parameters
+    ----------
+
+    image : turicreate.Image | SArray
+        The image or SArray of images to be resized.
+    min_side_length : int
+        The size of the minimum side of the image to be resized while maintaining aspect ratio.
+        For ex: 
+            if the image shape is : 720 * 960 i.e => the aspect ratio is 3:4
+            so if you set the min_side_length to 600, then the resized image dimension would be => 600 * 800
+    max_side_length : int
+        The size of the maximum side of the image to be resized while maintaining aspect ratio.
+        For ex: 
+            if the image shape is : 720 * 960 i.e => the aspect ratio is 3:4
+            so if you set the max_side_length to 600, then the resized image dimension would be => 450 * 600
+    channels : int, optional
+        The number of channels the image is resized to. 1 channel
+        corresponds to grayscale, 3 channels corresponds to RGB, and 4
+        channels corresponds to RGBA images.
+    decode : bool, optional
+        Whether to store the resized image in decoded format. Decoded takes
+        more space, but makes the resize and future operations on the image faster.
+    resample : 'nearest' or 'bilinear'
+        Specify the resampling filter:
+
+            - ``'nearest'``: Nearest neigbhor, extremely fast
+            - ``'bilinear'``: Bilinear, fast and with less aliasing artifacts
+
+    Returns
+    -------
+    out : turicreate.Image
```

Comment:
It's could also be an SArray, if the input is an SArray, right?

Reply:
Yes, you are right. I completely missed it. Will add SArray here